### PR TITLE
DAOS-9785 pool: check srv handle for ds_pool_tgt_query_map_handler

### PR DESCRIPTION
**Pull request by wangdi1**
Check server to server handle for ds_pool_tgt_query_map_handler for
map refresh process between servers.

Signed-off-by: Di Wang <di.wang@intel.com>

### CODE REVIEW

**Comment by liw** [comment]
@wangdi1, should I drop https://github.com/daos-stack/daos/pull/7082 then?

File: src/pool/srv_target.c [hunk 1]
```diff
@@ -1528,14 +1528,24 @@ ds_pool_tgt_query_map_handler(crt_rpc_t *rpc)
...
-		goto out;
+	if (daos_rpc_from_client(rpc) &&
+	    !is_pool_from_srv(in->tmi_op.pi_uuid, in->tmi_op.pi_hdl)) {
+		hdl = ds_pool_hdl_lookup(in->tmi_op.pi_hdl);
+		if (hdl == NULL) {
+			rc = -DER_NO_HDL;
+			goto out;
+		}
+		pool = hdl->sph_pool;
+	} else if (!daos_rpc_from_client(rpc) &&
```
Comment (kccain) [inline src/pool/srv_target.c:1539]: I am confused between looking at this patch, another similar patch https://github.com/daos-stack/daos/pull/7082
and the associated tickets https://daosio.atlassian.net/browse/DAOS-9785 and https://daosio.atlassian.net/browse/DAOS-8870

Where can the engine code be found that sends a POOL_TGT_QUERY_MAP RPC? I only see that in pool map refresh code in daos/src/pool/cli.c that I assume is not called by engine code.

**Comment by wangdi1** [comment]
> @wangdi1, should I drop #7082 then?

Ah, no, sorry, forgot your patch. I will close this one.

File: src/pool/srv_target.c (hunk 1 above)
Comment (wangdi1) [reply src/pool/srv_target.c:1539]: > Where can the engine code be found that sends a POOL_TGT_QUERY_MAP RPC? I only see that in pool map refresh code in daos/src/pool/cli.c that I assume is not called by engine code.

EC aggregation (on engine) needs to fetch the data on another engine by using object client stack, which will use map_refresh. see agg_fetch_odata_cells()

**Comment by wangdi1** [comment]
close this one, this is already resolved by PR-7082

**Pull request closed without merging**